internal/clients/team4/iigo.go

```diff
@@ -186,6 +179,41 @@ func (c *client) ResourceReport() shared.ResourcesReport {
 	return resReportStruct
 }
 
-////////////// TODO: FUNCTION WAITING ON BASECLIENT PR /////////////
-// GetTaxContribution()
+// GetTaxContribution gives value of how much the island wants to pay in taxes
+// The tax is the minimum contribution, you can pay more if you want to
+// COMPULSORY
+func (c *client) GetTaxContribution() shared.Resources {
+	valToBeReturned := shared.Resources(0)
+	currentWealth := c.ServerReadHandle.GetGameState().ClientInfo.Resources
+
+	collaborationThreshold := 1.0
+	wealthThreshold := 5 * valToBeReturned
+
+	// Initialise importance vector and parameters vector.
+	importance := mat.NewVecDense(6, []float64{
+		-2.0, -2.0, 0.0, 4.0, 0.0, 1.0,
+	})
```

Comment:
I'm not keen on having golden numbers in here. shouldn't those weights also be part of client configuration? So that we can tweak them easier in the construction instead of going through the code every time

Reply:
I agree. I am now adding them as a struct of importances, one for each function, that is initialised in "Initialise" inside client.go. Hope this is tidier to change when we're adjusting the parameters.